pkg/registry/core/shoot/strategy.go

```diff
@@ -101,6 +103,16 @@ func dropEnableBasicAuthenticationField(shoot *core.Shoot) {
 	}
 }
 
+// setKubernetesDashboardAuthMode sets the kubernetes-dashboard authentication mode to "token", if its current value is "basic".
+func setKubernetesDashboardAuthMode(shoot *core.Shoot) {
+	if shoot.Spec.Addons != nil && shoot.Spec.Addons.KubernetesDashboard != nil {
+		if authMode := shoot.Spec.Addons.KubernetesDashboard.AuthenticationMode; authMode != nil && *authMode == core.KubernetesDashboardAuthModeBasic {
```

Comment:
This way, you can drop the constant while still preserving functional correctness:

```suggestion
		if authMode := shoot.Spec.Addons.KubernetesDashboard.AuthenticationMode; authMode != nil && *authMode != KubernetesDashboardAuthModeToken {
```